src/main/kotlin/by/jprof/telegram/opinions/processors/YoutubeLinksProcessor.kt

```
@@ -0,0 +1,150 @@
+        regexGroups?.get(VIDEO_ID_GROUP_INDEX)?.let { videoId ->
+            logger.debug("Youtube video id is: $videoId")
+            val response = youTube.videos().list("snippet,statistics").setId(videoId).execute()
+
+            val videoDetails = response.items.first()
+            val snippet = videoDetails.snippet
+            val channelId = snippet.channelId
+            val title = snippet.title
+            val likes = videoDetails.statistics.likeCount
+            val dislikes = videoDetails.statistics.dislikeCount
+            val rawDescription = if (snippet.description.length > ACCEPTED_DISPLAY_LEN) {
+                snippet.description.substring(IntRange(0, ACCEPTED_DISPLAY_LEN)) + "..."
+            } else {
+                snippet.description
+            }
+
+            val description = rawDescription.escapeMarkdownV2Common()
+
+
+
+            runBlocking {
+                logger.debug("checking if $channelId is in a white list")
+                if (youtubeDAO.isInWhiteList(channelId)) {
+                    logger.debug("$channelId is in a white list")
+                    val videoText = "${"Vote for video: $youtubeLink".boldMarkdownV2()} \n$description " +
+                            "\nLikes: $likes Dislikes: $dislikes".boldMarkdownV2() //trim indent have strange layout
+                    val votes = getVotesByYoutubeId(videoId)
+                    bot.sendMessage(
+                            chatId = update.data.chat.id,
+                            text = videoText,
+                            parseMode = MarkdownV2ParseMode,
+                            replyToMessageId = update.data.messageId,
+                            replyMarkup = InlineKeyboardMarkup(keyboard = votingKeyBoard(votes, videoId))
+                    )
+                } else {
+                    logger.debug("$channelId is not in a white list")
+                    bot.sendMessage(
```

Let's just silently skip those videos?

---

yup, ok :)